int32 time
---
string message
